File: backend/WMS.UserManagement/Model/Product.cs

using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace WMS.UserManagement.Model
{
    [Table("product")]
    public class Product
    {
        [Column("product_id")]
        public string Id { get; set; }
        [Column("warehouse_id")]
        [Required]
        public Warehouse WarehouseId { get; set; }
        [Column("company_id")]
        [Required]
        public Company CompanyId { get; set; }
        [Column("uom_id")]
        [Required]
        public UnitOfMessure UnitOfMessureId { get; set; }
        [Column("product_name")]
        [Required]
        public string Name { get; set; }
        [Column("product_code")]
        [Required]
        public string Code { get; set; }
        [Column("product_description")]
        public string Description { get; set; }
        [Column("uom_created_on")]
        [DatabaseGenerated(DatabaseGeneratedOption.Computed)]
        public DateTime CreatedOn { get; }
        [Column("product_modified_on")]
        public DateTime ModifiedOn { get; }
    }
}

File: nodeBackend/DbConnectionModule/models/product.js
const Sequelize = require('sequelize');
module.exports = function(sequelize, DataTypes) {
  return sequelize.define('product', {
    product_id: {
      autoIncrement: true,
      type: DataTypes.INTEGER,
      allowNull: false,
      primaryKey: true
    },
    WarehouseId: {
      type: DataTypes.INTEGER,
      allowNull: true,
      references: {
        model: 'Warehouses',
        key: 'warehouse_id'
      }
    },
    CompanyId: {
      type: DataTypes.INTEGER,
      allowNull: true,
      references: {
        model: 'company',
        key: 'company_id'
      }
    },
    UnitOfMessureId: {
      type: DataTypes.INTEGER,
      allowNull: true,
      references: {
        model: 'uom',
        key: 'uom_id'
      }
    },
    product_name: {
      type: DataTypes.TEXT,
      allowNull: false
    },
    product_code: {
      type: DataTypes.TEXT,
      allowNull: false
    },
    product_description: {
      type: DataTypes.TEXT,
      allowNull: true
    }
  }, {
    sequelize,
    tableName: 'product',
    schema: 'dbo',
    timestamps: false,
    indexes: [
      {
        name: "IX_product_CompanyId",
        fields: [
          { name: "CompanyId" },
        ]
      },
      {
        name: "IX_product_UnitOfMessureId",
        fields: [
          { name: "UnitOfMessureId" },
        ]
      },
      {
        name: "IX_product_WarehouseId",
        fields: [
          { name: "WarehouseId" },
        ]
      },
      {
        name: "PK_product",
        unique: true,
        fields: [
          { name: "product_id" },
        ]
      },
    ]
  });
};
